<Type Name="TargetException" FullName="System.Reflection.TargetException">
  <Metadata><Meta Name="ms.openlocfilehash" Value="ceed8adc17d7854bc3e5a76c7a97eb88602ad4d6" /><Meta Name="ms.sourcegitcommit" Value="bac13a81377b1c4ebf3990e3287a6f1700f73751" /><Meta Name="ms.translationtype" Value="MT" /><Meta Name="ms.contentlocale" Value="fr-FR" /><Meta Name="ms.lasthandoff" Value="03/19/2019" /><Meta Name="ms.locfileid" Value="57962393" /></Metadata><TypeSignature Language="C#" Value="public class TargetException : ApplicationException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit TargetException extends System.ApplicationException" />
  <TypeSignature Language="DocId" Value="T:System.Reflection.TargetException" />
  <TypeSignature Language="VB.NET" Value="Public Class TargetException&#xA;Inherits ApplicationException" />
  <TypeSignature Language="C++ CLI" Value="public ref class TargetException : ApplicationException" />
  <TypeSignature Language="F#" Value="type TargetException = class&#xA;    inherit ApplicationException" />
  <AssemblyInfo>
    <AssemblyName>System.Reflection</AssemblyName>
    <AssemblyVersion>4.1.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>1.0.5000.0</AssemblyVersion>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>System.Runtime</AssemblyName>
    <AssemblyVersion>4.2.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.1.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Exception</BaseTypeName>
    <BaseTypeName FrameworkAlternate="netframework-4.5">System.ApplicationException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute FrameworkAlternate="netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0;netframework-4.8">
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
    <Attribute FrameworkAlternate="netframework-1.1;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0">
      <AttributeName>System.Serializable</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary><span data-ttu-id="f3936-101">Représente l’exception levée lors d’une tentative d’appel à une cible non valide.</span><span class="sxs-lookup"><span data-stu-id="f3936-101">Represents the exception that is thrown when an attempt is made to invoke an invalid target.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="f3936-102">Une `TargetException` est levée lors d'une tentative d'appel à une méthode non statique sur un objet nul.</span><span class="sxs-lookup"><span data-stu-id="f3936-102">A `TargetException` is thrown when an attempt is made to invoke a non-static method on a null object.</span></span> <span data-ttu-id="f3936-103">Cela peut se produire, car l’appelant n’a pas accès au membre ou la cible sans définir le membre et ainsi de suite.</span><span class="sxs-lookup"><span data-stu-id="f3936-103">This may occur because the caller does not have access to the member, or because the target does not define the member, and so on.</span></span>  
  
> [!NOTE]
>  <span data-ttu-id="f3936-104">Cette exception n’est pas incluse dans le [les applications .NET pour Windows Store](https://go.microsoft.com/fwlink/?LinkID=247912) ou [bibliothèque de classes Portable](~/docs/standard/cross-platform/cross-platform-development-with-the-portable-class-library.md), mais elle est levée par certains membres sont.</span><span class="sxs-lookup"><span data-stu-id="f3936-104">This exception is not included in the [.NET for Windows Store apps](https://go.microsoft.com/fwlink/?LinkID=247912) or the [Portable Class Library](~/docs/standard/cross-platform/cross-platform-development-with-the-portable-class-library.md), but it is thrown by some members that are.</span></span> <span data-ttu-id="f3936-105">Pour intercepter l’exception dans ce cas, vous devez écrire un `catch` instruction pour <xref:System.Exception> à la place.</span><span class="sxs-lookup"><span data-stu-id="f3936-105">To catch the exception in that case, write a `catch` statement for <xref:System.Exception> instead.</span></span>  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Exception" />
    <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="f3936-106">Gestion et levée des exceptions</span><span class="sxs-lookup"><span data-stu-id="f3936-106">Handling and Throwing Exceptions</span></span></related>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Reflection</AssemblyName>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary><span data-ttu-id="f3936-107">Initialise une nouvelle instance de la classe <see cref="T:System.Reflection.TargetException" />.</span><span class="sxs-lookup"><span data-stu-id="f3936-107">Initializes a new instance of the <see cref="T:System.Reflection.TargetException" /> class.</span></span></summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public TargetException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Reflection.TargetException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; TargetException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Reflection</AssemblyName>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="f3936-108">Initialise une nouvelle instance de la classe <see cref="T:System.Reflection.TargetException" /> à l’aide d’un message vide et de la cause première de l’exception.</span><span class="sxs-lookup"><span data-stu-id="f3936-108">Initializes a new instance of the <see cref="T:System.Reflection.TargetException" /> class with an empty message and the root cause of the exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="f3936-109">Le `InnerException` propriété est définie sur `null` et le code d’erreur HRESULT est COR_E_TARGET.</span><span class="sxs-lookup"><span data-stu-id="f3936-109">The `InnerException` property is set to `null` and the HRESULT error code is set to COR_E_TARGET.</span></span>  
  
 <span data-ttu-id="f3936-110">`TargetException` hérite de la <xref:System.Exception>.</span><span class="sxs-lookup"><span data-stu-id="f3936-110">`TargetException` inherits from the <xref:System.Exception>.</span></span> <span data-ttu-id="f3936-111">Ce constructeur définit les propriétés de la `Exception` de l’objet comme indiqué dans le tableau suivant.</span><span class="sxs-lookup"><span data-stu-id="f3936-111">This constructor sets the properties of the `Exception` object as shown in the following table.</span></span>  
  
|<span data-ttu-id="f3936-112">Property</span><span class="sxs-lookup"><span data-stu-id="f3936-112">Property</span></span>|<span data-ttu-id="f3936-113">Value</span><span class="sxs-lookup"><span data-stu-id="f3936-113">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|`null`|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="f3936-114">La chaîne vide ( » »).</span><span class="sxs-lookup"><span data-stu-id="f3936-114">The empty string ("").</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public TargetException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Reflection.TargetException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; TargetException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new System.Reflection.TargetException : string -&gt; System.Reflection.TargetException" Usage="new System.Reflection.TargetException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Reflection</AssemblyName>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="f3936-115"><see langword="String" /> décrivant la raison de l’exception.</span><span class="sxs-lookup"><span data-stu-id="f3936-115">A <see langword="String" /> describing the reason why the exception occurred.</span></span></param>
        <summary><span data-ttu-id="f3936-116">Initialise une nouvelle instance de la classe <see cref="T:System.Reflection.TargetException" /> à l’aide du message donné et de la cause première de l’exception.</span><span class="sxs-lookup"><span data-stu-id="f3936-116">Initializes a new instance of the <see cref="T:System.Reflection.TargetException" /> class with the given message and the root cause exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="f3936-117">`TargetException` hérite de la <xref:System.Exception>.</span><span class="sxs-lookup"><span data-stu-id="f3936-117">`TargetException` inherits from the <xref:System.Exception>.</span></span> <span data-ttu-id="f3936-118">Ce constructeur définit les propriétés de la `Exception` de l’objet comme indiqué dans le tableau suivant.</span><span class="sxs-lookup"><span data-stu-id="f3936-118">This constructor sets the properties of the `Exception` object as shown in the following table.</span></span>  
  
|<span data-ttu-id="f3936-119">Property</span><span class="sxs-lookup"><span data-stu-id="f3936-119">Property</span></span>|<span data-ttu-id="f3936-120">Value</span><span class="sxs-lookup"><span data-stu-id="f3936-120">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|`null`|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="f3936-121">Chaîne du message.</span><span class="sxs-lookup"><span data-stu-id="f3936-121">The message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected TargetException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Reflection.TargetException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; TargetException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="new System.Reflection.TargetException : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; System.Reflection.TargetException" Usage="new System.Reflection.TargetException (info, context)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Reflection</AssemblyName>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Security.SecuritySafeCritical</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" Index="0" FrameworkAlternate="netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netframework-1.1;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;netstandard-2.0;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" Index="1" FrameworkAlternate="netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netframework-1.1;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;netstandard-2.0;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0" />
      </Parameters>
      <Docs>
        <param name="info"><span data-ttu-id="f3936-122">Données pour la sérialisation ou la désérialisation de l’objet.</span><span class="sxs-lookup"><span data-stu-id="f3936-122">The data for serializing or deserializing the object.</span></span></param>
        <param name="context"><span data-ttu-id="f3936-123">Source et destination de l'objet.</span><span class="sxs-lookup"><span data-stu-id="f3936-123">The source of and destination for the object.</span></span></param>
        <summary><span data-ttu-id="f3936-124">Initialise une nouvelle instance de la classe <see cref="T:System.Reflection.TargetException" /> avec les informations de sérialisation et de contexte spécifiées.</span><span class="sxs-lookup"><span data-stu-id="f3936-124">Initializes a new instance of the <see cref="T:System.Reflection.TargetException" /> class with the specified serialization and context information.</span></span></summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public TargetException (string message, Exception inner);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception inner) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Reflection.TargetException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, inner As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; TargetException(System::String ^ message, Exception ^ inner);" />
      <MemberSignature Language="F#" Value="new System.Reflection.TargetException : string * Exception -&gt; System.Reflection.TargetException" Usage="new System.Reflection.TargetException (message, inner)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Reflection</AssemblyName>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="inner" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="f3936-125">Message d'erreur qui explique la raison de l'exception.</span><span class="sxs-lookup"><span data-stu-id="f3936-125">The error message that explains the reason for the exception.</span></span></param>
        <param name="inner"><span data-ttu-id="f3936-126">Exception ayant provoqué l'exception actuelle.</span><span class="sxs-lookup"><span data-stu-id="f3936-126">The exception that is the cause of the current exception.</span></span> <span data-ttu-id="f3936-127">Si le paramètre <paramref name="inner" /> n'est pas <see langword="null" />, l'exception actuelle est levée dans un bloc <see langword="catch" /> qui gère l'exception interne.</span><span class="sxs-lookup"><span data-stu-id="f3936-127">If the <paramref name="inner" /> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span></span></param>
        <summary><span data-ttu-id="f3936-128">Initialise une nouvelle instance de la classe <see cref="T:System.Reflection.TargetException" /> avec un message d'erreur spécifié et une référence à l'exception interne ayant provoqué cette exception.</span><span class="sxs-lookup"><span data-stu-id="f3936-128">Initializes a new instance of the <see cref="T:System.Reflection.TargetException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="f3936-129">Une exception levée en conséquence directe d'une exception précédente doit contenir une référence à celle-ci dans la propriété <xref:System.Exception.InnerException%2A>.</span><span class="sxs-lookup"><span data-stu-id="f3936-129">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="f3936-130">La propriété <xref:System.Exception.InnerException%2A> retourne la même valeur que celle qui a été passée dans le constructeur ou retourne `null` si la propriété <xref:System.Exception.InnerException%2A> ne fournit pas la valeur d'exception interne au constructeur.</span><span class="sxs-lookup"><span data-stu-id="f3936-130">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="f3936-131">Le tableau suivant montre les valeurs initiales des propriétés d'une instance de <xref:System.Reflection.TargetException>.</span><span class="sxs-lookup"><span data-stu-id="f3936-131">The following table shows the initial property values for an instance of <xref:System.Reflection.TargetException>.</span></span>  
  
|<span data-ttu-id="f3936-132">Property</span><span class="sxs-lookup"><span data-stu-id="f3936-132">Property</span></span>|<span data-ttu-id="f3936-133">Value</span><span class="sxs-lookup"><span data-stu-id="f3936-133">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="f3936-134">La référence à l'exception interne.</span><span class="sxs-lookup"><span data-stu-id="f3936-134">The inner exception reference.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="f3936-135">Chaîne du message d'erreur.</span><span class="sxs-lookup"><span data-stu-id="f3936-135">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Exception" />
        <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="f3936-136">Gestion et levée d’exceptions dans .NET</span><span class="sxs-lookup"><span data-stu-id="f3936-136">Handling and throwing exceptions in .NET</span></span></related>
      </Docs>
    </Member>
  </Members>
</Type>